### PR TITLE
fix(ivy): ensure @animation host bindings/listeners work properly

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/instructions.ts [hunk 1]
```diff
@@ -987,12 +987,45 @@ export function elementAttribute(
...
+}
+
+/**
+ * Update a property on an element using the current elements component's renderer.
+ *
+ * This behaves exactly the same as `elementProperty`, but it will only use the renderer
+ * of the sub component of the target element (which is the element derived from the provided
+ * `index` value).
+ *
+ * Note that this instruction will only work if and when the target element is the host element
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: This text needs a bit of word-smithing.

File: packages/core/src/render3/instructions.ts [hunk 2]
```diff
@@ -987,12 +987,45 @@ export function elementAttribute(
...
+ * (this is necessary for host property bindings)
+ */
+export function elementPropertyUseComponentRenderer<T>(
+    index: number, propName: string, value: T | NO_CHANGE, sanitizer?: SanitizerFn | null,
+    nativeOnly?: boolean) {
+  if (value === NO_CHANGE) return;
+  const lView = getLView();
+  const tNode = getTNode(index, lView);
+  const componentLView = lView[tNode.index] as LView;
+  elementPropertyInternal(componentLView, index, propName, value, sanitizer, nativeOnly);
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: This can't be right. The `index` is tide to `LView` and when you pass in a different `LView` it will not work. If tests pass it is only coincidence.

**Review by mhevery — CHANGES_REQUESTED** [review]

File: packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts [hunk 3]
```diff
@@ -281,8 +281,8 @@ describe('compiler compliance: styling', () => {
           template: function MyComponent_Template(rf, ctx) {
             if (rf & 1) {
               $r3$.ɵelementStart(0, "div", _c0);
-              $r3$.ɵlistener("@myAnimation.start", function MyComponent_Template_div__myAnimation_start_listener($event) { return ctx.onStart($event); });
-              $r3$.ɵlistener("@myAnimation.done", function MyComponent_Template_div__myAnimation_done_listener($event) { return ctx.onDone($event); });
+              $r3$.ɵlistener("@myAnimation.start", function MyComponent_Template_animation_myAnimation_start_listener($event) { return ctx.onStart($event); });
```
Comment (mhevery) [inline packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: Why did you change these names. I think the previous names were correct.

File: packages/core/src/render3/instructions.ts [hunk 4]
```diff
@@ -987,10 +987,48 @@ export function elementAttribute(
...
+ *
+ * @param index The index of the element to update in the data array
+ * @param propName Name of property. Because it is going to DOM, this is not subject to
+ *        renaming as part of minification.
+ * @param value New value to write.
+ * @param sanitizer An optional function used to sanitize the value.
+ * @param nativeOnly Whether or not we should only set native properties and skip input check
+ * (this is necessary for host property bindings)
+ */
+export function componentHostSyntheticProperty<T>(
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: Thanks for rewriting the comments. These are way better!

**Review by mhevery — CHANGES_REQUESTED** [review]
Some minor clean up 19fdeb8f18.

Overall this is great, just one small nit about the function name.

**Review by mhevery — APPROVED** [review]

**Comment by kara** [comment]
@matsko Can you rebase this? :)

**Comment by matsko** [comment]
@kara the approval didn't carry-over into the rebased state. We should be good to regardless.

**Comment by matsko** [comment]
https://tap.corp.google.com/ui#id=OCL:227787010:BASE:227857777:1546618018364:2f1dcf85
